test/images/nettest/nettest.go

```diff
@@ -241,11 +241,22 @@ func contactOthers(state *State) {
 	if err != nil {
 		log.Fatalf("Unable to create client; error: %v\n", err)
 	}
-	// Double check that worked by getting the server version.
-	if v, err := client.Discovery().ServerVersion(); err != nil {
-		log.Fatalf("Unable to get server version: %v\n", err)
-	} else {
-		log.Printf("Server version: %#v\n", v)
+
+	// Try to get the server version until <timeout>; we use a timeout because
+	// the pod might not have immediate network connectivity.
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(sleepTime) {
+		// Double check that worked by getting the server version.
+		if v, err := client.Discovery().ServerVersion(); err != nil {
+			log.Printf("Unable to get server version: %v\n", err)
+		} else {
+			log.Printf("Server version: %#v\n", v)
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	if err != nil {
```

Comment:
The `err` here does not reference the same `err` from inside the `for` loop, right? Sorta like in here https://play.golang.org/p/HpKlN7sGgl6

Reply:
That's a good point. Fixed.